plugins/nginx-vhosts/functions

```diff
@@ -61,29 +67,63 @@ configure_nginx_ports() {
   local RAW_TCP_PORTS="$(get_app_raw_tcp_ports "$APP")"
   local DOKKU_NGINX_PORT=$(config_get "$APP" DOKKU_NGINX_PORT)
   local DOKKU_NGINX_SSL_PORT=$(config_get "$APP" DOKKU_NGINX_SSL_PORT)
+  local DOKKU_PROXY_PORT_MAP=$(config_get "$APP" DOKKU_PROXY_PORT_MAP)
   local IS_APP_VHOST_ENABLED="$(is_app_vhost_enabled "$APP")"
 
```

Comment:
Most of the automagic/batteries included stuff is here wrt handling the default case and state transition from non-ssl to ssl and vice versa